### PR TITLE
fix: validate artifact uri from component recipe

### PR DESCRIPTION
**Pull request by abanthiy**
**Issue #, if available:**
Artifact Uri in component recipe was not being validated leading to NPE if the URI did not have a scheme.
**Description of changes:**
Adding the validation
**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 - [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java [hunk 1]
```diff
@@ -108,22 +111,34 @@ public void checkDownloadPrerequisites(List<ComponentArtifact> artifacts, List<C
...
                 }
+            } catch (InvalidArtifactUriException e) {
+                throw new PackageLoadingException(String
+                        .format("Failed to download due to bad artifact URI: %s", artifact.getArtifactUri()), e);
             }
         }
     }
+
+    private void validateArtifactUri(ComponentArtifact artifact) throws InvalidArtifactUriException {
+        if (artifact.getArtifactUri() == null) {
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java:134]: Add the component name/version to the error message

Comment (abanthiy) [reply src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java:134]: With the current method signature that mapping is not available. ComponentArtifact does not store the componentId/name.

Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java:134]: Is it possible to add? I can foresee that that will be confusing without knowing what component has the issue

**Review by MikeDombo — APPROVED** [review]

**Comment by MikeDombo** [comment]
What are we doing with this PR?

**Review by rbattle — APPROVED** [review]

**Pull request merged**